llvm/lib/Target/SystemZ/SystemZAsmPrinter.cpp

```diff
@@ -1541,6 +1540,40 @@ void SystemZAsmPrinter::emitPPA2(Module &M) {
   OutStreamer->popSection();
 }
 
+void SystemZAsmPrinter::emitGlobalAlias(const Module &M,
+                                        const GlobalAlias &GA) {
```

Comment:
Why is this necessary?  This seems unrelated to the above change - there was no GlobalAlias involved as far as I can see.  Also, there doesn't appear to be any test case verifying what (if any) special handling is needed for GlobalAlias.  (B.t.w. is that special handling actually z/OS related, or is it rather GOFF related?  If the latter - there is XCOFF specific handling in emitGlobalAlias, so maybe the GOFF related special handling needs to go there too?)

Reply:
This is required to complete the implementation for handling indirect symbols. In the case where there is a globalAlias, we require an external reference symbol with the indirect flag set which is handled by emitGlobalAlias.
I added a test for this case.